File: packages/core-types/src/results.ts
/* ============================================================================
 * Siragugal Film Studio
 * Copyright (C) 2026 Siragugal Film Studio Contributors
 * Licensed under Apache-2.0 or MIT.
 * ============================================================================ */

import { SiraError } from './errors';

export type AsyncJobStatus = 'PENDING' | 'RUNNING' | 'PAUSED' | 'CANCELLED' | 'COMPLETED' | 'FAILED';

export type SiraResult<T> =
  | { status: 'SUCCESS'; data: T }
  | { status: 'PARTIAL_SUCCESS'; data: T; warnings: SiraError[] }
  | { status: 'ERROR'; error: SiraError }
  | { status: 'PROGRESS'; progress: number; stage: string }
  | { status: 'CANCELLED'; reason: string };

File: packages/sira-types/src/results.rs
/* ============================================================================
 * Siragugal Film Studio
 * Copyright (C) 2026 Siragugal Film Studio Contributors
 * Licensed under Apache-2.0 or MIT.
 * ============================================================================ */

use crate::errors::SiraError;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
pub enum SiraResult<T> {
    Success(T),
    PartialSuccess { data: T, warnings: Vec<SiraError> },
    Error(SiraError),
    Progress { progress: f32, stage: String },
    Cancelled { reason: String },
}
